packages/vue-i18n-core/src/types.ts

```diff
@@ -18,28 +16,12 @@ export type Disposer = () => void
  *
  * declare module '@intlify/vue-i18n-core' {
  *   interface GeneratedTypeConfig {
- *     legacy: false
  *   }
  * }
  * ```
  */
 
 export interface GeneratedTypeConfig {}
 
-/**
- * Narrowed i18n instance type based on `GeneratedTypeConfig['legacy']`
- *
- * - `never` (unset) resolves to `VueI18n | ExportedGlobalComposer`
- * - `true` resolves to `VueI18n`
- * - `false` resolves to `ExportedGlobalComposer`
- */
-export type GeneratedInstanceType =
-  GeneratedTypeConfig extends Record<'legacy', infer Legacy> ? Legacy : never
-
 /** @VueI18nGeneral */
-export type VueI18nInstance =
-  IsNever<GeneratedInstanceType> extends true
-    ? VueI18n | ExportedGlobalComposer
-    : GeneratedInstanceType extends true
-      ? VueI18n
-      : ExportedGlobalComposer
+export type VueI18nInstance = ExportedGlobalComposer
```

Comment:
@BobbieGoede 
I think this change will be affected nuxt-i18n.
Can you check whether this change is okay with nuxt-i18n? 🙏

Reply:
Since these changes are for the next major of vue-i18n removing it should be fine!

We're actually still not yet (functionally) making use of the conditional type in nuxt-i18n v9 (uses vue-i18n v10), but we will in nuxt-i18n v10 (with vue-i18n v11).